broker-auth-multiuser-otp/src/main/java/org/aktin/broker/auth/otp/repository/FsUserRepository.java

```diff
@@ -0,0 +1,188 @@
+package org.aktin.broker.auth.otp.repository;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import java.util.logging.Logger;
+import javax.inject.Singleton;
+
+@Singleton
+public class FsUserRepository implements UserRepository {
+
+  private static final Logger log = Logger.getLogger(FsUserRepository.class.getName());
+
+  private static final String PROPERTY_USER_FILE = "aktin.broker.users.file";
+  private static final String DEFAULT_USER_FILE = "users.txt";
+  private static final String FIELD_SEPARATOR = ";";
```

Comment:
Dont use ; instead use 'tab' because this is more unlikely to occur inside a password or the hash algorithm. Otherwise this will be a security issue due to the lack of escaping.